File: 70-ClimbingStairs/70-ClimbingStairs.py
# Last updated: 9/14/2025, 9:17:48 AM
class Solution:
    def climbStairs(self, n: int) -> int:
        if n==1:
            return 1
        if n==2:
            return 2
        a,b,c=1,2,0
        for i in range(3,n+1):
            c=a+b
            b=c
            a=c-a
        return c

File: 70-ClimbingStairs/70-ClimbingStairs.java
// Last updated: 9/14/2025, 9:17:44 AM
class Solution {
    public int climbStairs(int n) {
        if(n==1){
            return 1;
        }
        if(n==2){
            return 2;
        }
        int arr[]=new int[n+1];
        arr[0]=1;
        arr[1]=1;
        for(int i=2;i<=n;i++){
            arr[i]=arr[i-1]+arr[i-2];
        }
        return arr[n];
}
}